### PR TITLE
Add cloudformation template for pattern-1 with puppet.

### PR DESCRIPTION
**Pull request by malithie**
## Purpose
This adds a cloudformation template for WSO2 Identity Server single node deployment with external datasources.
Deployment is done over a puppet master agent setup.

## Test environment
JDK: jdk-8u131 
DataSource: MYSQL
WSO2 IS 5.4.0

### CODE REVIEW

File: cloudformation-templates/pattern-1/pattern-1-with-puppet-cloudformation.template.yml [hunk 1]
```diff
@@ -0,0 +1,603 @@
...
+  WSO2EnvInstanceSecurityGroup:
+    Type: 'AWS::EC2::SecurityGroup'
+    Properties:
+      GroupDescription: Security group for IS EC2 compute instances
+      VpcId: !Ref WSO2EnvVPC
+      SecurityGroupIngress:
+      - IpProtocol: tcp
+        FromPort: 0
+        ToPort: 65535
+        CidrIp: 0.0.0.0/0
```
Comment (chamilad) [inline cloudformation-templates/pattern-1/pattern-1-with-puppet-cloudformation.template.yml]: Is it required to open all ports to outside for the instance security groups?

File: cloudformation-templates/pattern-1/pattern-1-with-puppet-cloudformation.template.yml [hunk 2]
```diff
@@ -0,0 +1,603 @@
...
+  WSO2EnvDBSecurityGroup:
+      Type: 'AWS::EC2::SecurityGroup'
+      Properties:
+        GroupDescription: Security group for IS RDS
+        VpcId: !Ref WSO2EnvVPC
+        SecurityGroupIngress:
+        - IpProtocol: tcp
+          FromPort: 3306
+          ToPort: 3306
+          CidrIp: 0.0.0.0/0
```
Comment (chamilad) [inline cloudformation-templates/pattern-1/pattern-1-with-puppet-cloudformation.template.yml]: Isn't it better to restrict this to connections from the instance security group?

**Review by chamilad — CHANGES_REQUESTED** [review]

**Review by chamilad — APPROVED** [review]
After an offline discussion with @malithie it was decided to push the suggested changes to a later PR for the purpose of ease of testing.

**Comment by malithie** [comment]
Fixed review comments from @chamilad

**Review by isharak — APPROVED** [review]

**Pull request merged**